src/App.js

```diff
@@ -21,19 +23,166 @@ const firebaseConfig = {
+                // This gives you a Google Access Token. You can use it to access the Google API.
+                const credential = GoogleAuthProvider.credentialFromResult(result);
+                const token = credential.accessToken;
+
+                // The signed-in user info.
+                const user = result.user;
+                setUser(user);
+                console.log("Signed In!");
+                console.log(user);
+                setPage({type: "home"});
+                // ...
+            }).catch((error) => {
+                // Handle Errors here.
+                const errorCode = error.code;
+                const errorMessage = error.message;
+                // The email of the user's account used.
+                const email = error.email;
+                // The AuthCredential type that was used.
+                const credential = GoogleAuthProvider.credentialFromError(error);
+                console.log("try signing in again!");
+                setValidity(false);
+                // ...
+        });
+    }
+
+    function handleSignOut(){
+        setPage({type: "trueHome"});
+    }
+
+    const pageRenderLookup = {
+        "signIn": (
+            <div className={"home-page"}>
+                <img tabIndex="0" onKeyPress={(event) => {(event.key === "Enter"||event.code === "Space") && setPage({type: "trueHome"})}} onClick={() => setPage({type: "trueHome"})} alt={"Back Arrow"} src={"long-arrow-alt-left-solid.svg"} className={"back-arrow"}/>
+                <h1 className={"sign-in-up-h1s"}>Sign into your Account</h1>
+                {!passwordValid && <p className={"wrongPass"}>Wrong username or password</p>}
+                <div className={"buttons-box"}>
+                    <div className={"authbuttons"}>
+                        <input className="authitems" placeholder={"Email"} type="text" name="email" onChange={(e) => setEmail(e.target.value)}/>
+                        <input className="authitems" placeholder={"Password"} type="password" name="password" onChange={(e) => setPassword(e.target.value)}/>
+                        <button className="front-page-buttons" onClick={() => handleSignIn(userEmail, userPassword)}> Sign In </button>
+                        <p className={"authOption"}>or</p>
+                        <button className="front-page-buttons" onClick={() => handleGoogleSignIn()}>Sign In with Google</button>
+                    </div>
+                </div>
+            </div>
+        ),
+        "signUp": (
+            <div className={"home-page"}>
+                <img tabIndex="0" onKeyPress={(event) => {(event.key === "Enter"||event.code === "Space") && setPage({type: "trueHome"})}} onClick={() => setPage({type: "trueHome"})} alt={"Back Arrow"} src={"long-arrow-alt-left-solid.svg"} className={"back-arrow"}/>
+                <h1 className={"sign-in-up-h1s"}>Create an Account</h1>
+                {!signupValid && <p className={"wrongPass"}>Invalid email or password</p>}
```

Comment:
Hardcoding error message rather than seeing the actual error (could be, among other things: invalid email address, password too short, account already exists, etc.)